lib/checks/aria/allowed-attr.js

```diff
@@ -6,7 +6,7 @@ var attr,
 	attrName,
 	allowed,
 	role = node.getAttribute('role'),
-	attrs = node.attributes;
+	attrs = axe.utils.getNodeAttributes(node);
```

Comment:
alternatively, add a custom eslint rule, which will stop people from using node.attributes, because it is hard to remember that a util method exists.

Something like this could help - https://github.com/dequelabs/axe-core/blob/develop/.eslintrc#L71

Reply:
Nice, I like that.

Reply:
I'm not sure about this. Adding an ESLint rule which prevents accessing standard DOM properties seems a bit arbitrary.

There are two major reasons I'm against it:

1. the rule will confuse people who aren't aware of it, thus making `axe-core` an even more difficult project to contribute to
2. it promotes adding more rules like it, which will make the first point even worse

Before we build a new rule, can we come to a consensus here and make sure it's the right decision?

Reply:
I don't know. If the result of accessing the standard DOM property is potential bugs, I see that fitting under the realm of eslint.

Reply:
I understand that it'll potentially prevent bugs, and that's why I'm not saying "no". I just want be sure it's worth the risk.

Can we be sure that this rule doesn't create false positives? If not, I'm of the opinion this will add *a lot* of confusion down the line.

What if a future developer does this:

```js
class NodeImpl {
  attributes = 7
}

var node = new NodeImpl()
var atrrs = node.attributes
```

Will the ESLint rule fail their build?

Reply:
To help with user understanding, we can add a comment above the eslint rule that links back to this pr:

```js
{
  // node.attributes can be clobbered so is unsafe to use
  // @see https://github.com/dequelabs/axe-core/pull/1432 
  "selector": "MemberExpression[object.name=node][property.name=attributes]",
  "message": "Don't use node.attributes use node.hasAttirbutes() or axe.utils.getNodeAttributes(node) instead."
}
``

Reply:
As discussed on our call, I'm OK with adding the ESLint thing along with the comment. If it becomes an issue in the future, we can revisit it then. Enough yak hair has been removed here.